### PR TITLE
🐛 Fix metadata for e2e upgrade tests

### PR DESCRIPTION
**Pull request by mboukhalfa**
Adding metadata to upgrade tests based on the upgrade from version.
Fix root metadata by removing the v1.0 because it does not exists on capm3 releases

### CODE REVIEW

File: test/e2e/data/shared/v1.3/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  minor: 3
+  contract: v1beta1
+- major: 1
+  minor: 2
+  contract: v1beta1
+- major: 1
+  minor: 1
+  contract: v1beta1
+- major: 1
+  minor: 0
```
Comment (kashifest) [inline test/e2e/data/shared/v1.3/metadata.yaml:14]: remove this also

**Review by kashifest — CHANGES_REQUESTED** [review]
please update https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/test/e2e/data/shared/v1.4/metadata.yaml as well. One more change requested inline.

**Comment by kashifest** [comment]
> _ No description provided. _

File: test/e2e/data/shared/v1.3/metadata.yaml (hunk 1 above)
Comment (mboukhalfa) [reply test/e2e/data/shared/v1.3/metadata.yaml:14]: this capi metadata

**Review by kashifest — APPROVED** [review]

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**